contracts/ethereum/contracts/Registry.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: GPL
+
+pragma solidity ^0.8.11;
+pragma abicoder v2;
+
+import "./IToken.sol";
+import "./ILiquidityProvider.sol";
+import "./IToken.sol";
+import "./IRegistry.sol";
+
+import "@openzeppelin/contracts/proxy/beacon/IBeacon.sol";
+
+contract Registry is IRegistry {
+
+    // @dev RegistrationType is a uint8 in practice, so it can be updated
+    // with a contract upgrade if the ABI changes
+    event RegistrationMade(RegistrationType type_, address indexed addr);
+
+    uint8 version_;
+
+    address operator_;
+
+    /// @notice tokenBeacon_ to be used as a helpful guide for the DAO
+    IBeacon tokenBeacon_;
+
+    /// @notice liquidityProviderBeacon_ as a helpful guide for the DAO
+    IBeacon liquidityProviderBeacon_;
+
+    Registration[] registrations_;
```

Comment:
The intention was to not require updating if the ABI changes since you can add something to the enum and then just add to it and the consumers will assume the type is there